pkg/cmd/monitor_command/run_monitor_command.go

```diff
@@ -0,0 +1,181 @@
+		Use:   "run-monitor",
+		Short: "Continuously verify the cluster is functional",
+		Long: templates.LongDesc(`
+		Run a continuous verification process
+
+		`),
+
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return monitorOpt.Run()
+		},
+	}
+	cmd.Flags().StringVar(&monitorOpt.ArtifactDir,
+		"artifact-dir", monitorOpt.ArtifactDir,
+		"The directory where monitor events will be stored.")
+	return cmd
+}
+
+// Run starts monitoring the cluster by invoking Start, periodically printing the
+// events accumulated to Out. When the user hits CTRL+C or signals termination the
+// condition intervals (all non-instantaneous events) are reported to Out.
+func (opt *RunMonitorOptions) Run() error {
+	ctx, cancelFn := context.WithCancel(context.Background())
+	defer cancelFn()
+	abortCh := make(chan os.Signal, 2)
+	go func() {
+		<-abortCh
+		fmt.Fprintf(opt.ErrOut, "Interrupted, terminating\n")
+		cancelFn()
+		sig := <-abortCh
+		fmt.Fprintf(opt.ErrOut, "Interrupted twice, exiting (%s)\n", sig)
+		switch sig {
+		case syscall.SIGINT:
+			os.Exit(130)
+		default:
+			os.Exit(0)
+		}
+	}()
+	signal.Notify(abortCh, syscall.SIGINT, syscall.SIGTERM)
+
+	restConfig, err := monitor.GetMonitorRESTConfig()
+	if err != nil {
+		return err
+	}
+	m, err := monitor.Start(ctx, restConfig, opt.AdditionalEventIntervalRecorders)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		ticker := time.NewTicker(100 * time.Millisecond)
+		defer ticker.Stop()
+		var last time.Time
+		done := false
+		for !done {
+			select {
+			case <-ticker.C:
+			case <-ctx.Done():
+				done = true
+			}
+			events := m.Intervals(last, time.Time{})
+			if len(events) > 0 {
+				for _, event := range events {
+					if !event.From.Equal(event.To) {
+						continue
+					}
+					fmt.Fprintln(opt.Out, event.String())
+				}
+				last = events[len(events)-1].From
+			}
+		}
+	}()
+
+	<-ctx.Done()
+
+	time.Sleep(150 * time.Millisecond)
+	if events := m.Conditions(time.Time{}, time.Time{}); len(events) > 0 {
+		fmt.Fprintf(opt.Out, "\nConditions:\n\n")
+		for _, event := range events {
+			fmt.Fprintln(opt.Out, event.String())
+		}
+	}
+
+	// Store events to artifact directory
+	if len(opt.ArtifactDir) != 0 {
+		recordedEvents := m.Intervals(time.Time{}, time.Time{})
+		recordedResources := m.CurrentResourceState()
+		timeSuffix := fmt.Sprintf("_%s", time.Now().UTC().Format("20060102-150405"))
+
+		eventDir := fmt.Sprintf("%s/monitor-events", opt.ArtifactDir)
+		if err := os.MkdirAll(eventDir, os.ModePerm); err != nil {
+			fmt.Printf("Failed to create monitor-events directory, err: %v\n", err)
+			return err
+		}
+
+		if err := monitor.WriteEventsForJobRun(eventDir, recordedResources, recordedEvents, timeSuffix); err != nil {
+			fmt.Printf("Failed to write event data, err: %v\n", err)
+			return err
+		}
+		if err := monitor.WriteTrackedResourcesForJobRun(eventDir, recordedResources, recordedEvents, timeSuffix); err != nil {
```

Comment:
Hm..., you are now calling `WriteTrackedResourcesForJobRun` here.  I see it's also called [here](https://github.com/openshift/origin/blob/88793278dda83b5fd5703dc8eea329f806478c20/pkg/test/ginkgo/options_monitor_events.go#L75)
Will the call on line 148 overwrite what the call in line 75 already generated?

Reply:
> Hm..., you are now calling `WriteTrackedResourcesForJobRun` here. I see it's also called [here](https://github.com/openshift/origin/blob/88793278dda83b5fd5703dc8eea329f806478c20/pkg/test/ginkgo/options_monitor_events.go#L75)
> Will the call on line 148 overwrite what the call in line 75 already generated?

those are two different types I think.  I'd be happy to see them collapsed into one, but I don't think we need that in order to get value here.